src/clj/fluree/db/query/exec.cljc

```diff
@@ -42,22 +43,85 @@
     (async/take 1 result-ch)
     (async/into [] result-ch)))
 
+(defn extract-subquery
+  [query-smt]
+  (when (and (sequential? query-smt)
+             (= :query (first query-smt)))
+    (second query-smt)))
+
+(defn execute
+  [ds fuel-tracker q error-ch initial-soln]
+  (->> (where/search ds q fuel-tracker error-ch initial-soln)
```

Comment:
I think there's some room to consolidate everything before the format step into it's own function, and to consolidate the offset and limit steps into another function too since so much logic is shared between the `execute` and `execute-subquery` functions.

Reply:
I wanted to consolidate these more, but the logic *is* different, even though some functions are shared. Wrapping these in multiple steps IMO makes it harder to follow the logic and right now I think they both read very clearly.

I can change if you feel differently, but I did go down that path and reverted back here.